### PR TITLE
[Core] Unset RAY_RAYLET_NODE_ID for long running nightly tests 

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyyx <rickyx@anyscale.com>

## Why are these changes needed?
For long running tests that has `ray stop` as part of the prepare steps, the node started in the test will re-use the node id. 

More details in this #28562 

## Testing Plan
I will trigger some of these long running tests just to make sure they could start (and kill afterwards)

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [X] Unit tests
   - [ ] Release tests:
   - [X] This PR is not tested :(

### CODE REVIEW

**Comment by rickyyx** [comment]
Release tests: https://buildkite.com/ray-project/release-tests-pr/builds/15809

**Comment by stephanie-wang** [comment]
Seems like it might not have worked? Maybe we need to run it as part of the release test instead of in the app env, e.g., modify [this line](https://github.com/ray-project/ray/blob/f2568e14e7677e9c07688f8069854e818d23e189/release/release_tests.yaml#L1856).

**Comment by scv119** [comment]
tests are failing

**Review by stephanie-wang — APPROVED** [review]

File: release/long_running_tests/app_config.yaml [hunk 1]
```diff
@@ -20,4 +20,4 @@ post_build_cmds:
   # TODO (Alex): Ideally we would install all the dependencies from the new
   # version too, but pip won't be able to find the new version of ray-cpp.
   - pip3 uninstall ray -y && pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}
-  - {{ env["RAY_WHEELS_SANITY_CHECK"] | default("echo No Ray wheels sanity check") }}
+  - {{ env["RAY_WHEELS_SANITY_CHECK"] | default("echo No Ray wheels sanity check") }}
```
[the remark this answers is not in the document]
Comment (rickyyx) [reply release/long_running_tests/app_config.yaml:23]: oops - double DDed it.

**Comment by rickyyx** [comment]
@scv119 @stephanie-wang @iycheng Tests are running - should be good to merge. https://buildkite.com/ray-project/release-tests-pr/builds/15922#_

**Comment by scv119** [comment]
great catch!

**Comment by rickyyx** [comment]
Cancelled all long running tests.

**Pull request merged**